crates/hc/src/bin/hc.rs

```diff
@@ -0,0 +1,237 @@
+    #[structopt(subcommand)]
+    op: Op,
+    /// Force the admin port to a specific value.
+    /// Useful if you are setting this config
+    /// up for use elsewhere (see also secondary_admin_port).
+    /// For example `hc -f=9000,9001`
+    #[structopt(short, long, value_delimiter = ",")]
+    force_admin_ports: Vec<u16>,
+    /// Set the path to the holochain binary.
+    #[structopt(short, long, env = "HC_HOLOCHAIN_PATH", default_value = "holochain")]
+    holochain_path: PathBuf,
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = structopt::clap::AppSettings::InferSubcommands)]
+enum Op {
+    /// Generate a fresh holochain setup.
+    Generate {
+        #[structopt(flatten)]
+        gen: Create,
+        #[structopt(flatten)]
+        source: Source,
+    },
+    /// Run a conductor from existing or new setup.
+    Run(Run),
+    /// Call conductor admin interfaces.
+    Call(hc::calls::Call),
+    /// [WIP unimplemented]: Run custom tasks using cargo task
+    Task,
+    /// List setups found in `$(pwd)/.hc`.
+    /// Note that indices will change if setups are removed.
+    List {
+        /// Show more verbose information.
+        #[structopt(short, long, parse(from_occurrences))]
+        verbose: usize,
+    },
+    /// Clean setups that are listed in the current directory
+    /// inside the `.hc` file.
+    Clean {
+        /// Setups to clean.
+        /// If blank all will be cleaned.
+        setups: Vec<usize>,
+    },
+}
+
+#[derive(Debug, StructOpt)]
+struct Run {
+    #[structopt(short, long, value_delimiter = ",")]
+    /// Optionally create an app interface.
+    /// This allows you UI to talk to the conductor.
+    /// For example `hc run -p=0,9000,0`
+    ports: Vec<u16>,
+    #[structopt(short, long, value_delimiter = ",")]
+    /// Specify a secondary admin port so you can make admin requests while holochain is running.
+    /// For example: `hc run -s=0,9000,1`
+    /// Set to 0 to let OS choose a free port.
+    /// Defaults to all 0.
+    /// Set `disable-secondary` to disable this.
+    secondary_admin_ports: Vec<u16>,
+    #[structopt(short, long)]
+    disable_secondary: bool,
+    /// Generate a new setup and then run.
+    #[structopt(subcommand)]
+    gen: Option<Gen>,
+    #[structopt(flatten)]
+    existing: Existing,
+    #[structopt(flatten)]
+    source: Source,
+}
+
+#[derive(Debug, StructOpt)]
+enum Gen {
+    Gen(Create),
+}
+
+#[derive(Debug, StructOpt)]
+struct Source {
+    #[structopt(short, long, default_value = "1")]
+    /// Number of conductors to create or run.
+    /// Must be <= to number of existing setups.
+    /// Defaults to 1.
+    num_conductors: usize,
+    /// List of dnas to run.
+    /// Defaults to the current directory
+    dnas: Vec<PathBuf>,
+}
+
+impl Gen {
+    pub fn into_inner(self) -> Create {
+        match self {
+            Gen::Gen(n) => n,
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    observability::test_run().ok();
```

Comment:
not a big deal, but maybe we need an api in observability that doesn't include the word 'test' so it's not confusing : )

Reply:
Yeh there is one I was just being lazy